src/sorting.jl

```diff
@@ -360,7 +360,7 @@ function quicksort!(c::AbstractArray{T}; lt::F1, by::F2) where {T,F1,F2}
 
     kernel = @cuda launch=false qsort_kernel(c, 0, N, true, Val(MAX_DEPTH > 1), MAX_DEPTH, nothing, lt, by)
 
-    get_shmem(threads) = threads * (sizeof(Int32) + max(4, sizeof(T)))
```

Comment:
@xaellison why did you clamp to `4` here? The other `shmem` calculations (in the kernel) didn't do that.

Reply:
Off the top of my head, I can't remember, but that seems wrong. `get_shmem(threads) = threads * (sizeof(Int32) + sizeof(T))` would be consistent